apis/common/v1/connection_details.go

```diff
@@ -68,11 +67,11 @@ type ConnectionSecretMetadata struct {
 }
 
 // SetOwnerUID sets owner object uid label.
-func (in *ConnectionSecretMetadata) SetOwnerUID(so metav1.Object) {
+func (in *ConnectionSecretMetadata) SetOwnerUID(uid string) {
```

Comment:
Nit: We could take a `types.UID` here too, though I guess at some point we need to convert it to a `string` anyhow.